### ISSUE TITLE
240p video output issues with new video driver code.

### ISSUE THREAD

**Issue by alphanu1** [libretro/RetroArch#8366]
@twinaphex @bparker06 It seems since the new addition of set geometry, video output at 240p has lost stability.

This is with and without CRTSwitchRes. 

Before the new video driver implementation Retroarch did not change any X11 setting that caused an X refresh.

So, now even users that only want a 240p output without resolution switching are suffering this issue as well as people with 31khz CRT PC monitors.

Please look at these videos for comparison.

Before new video driver code.
https://youtu.be/Mi5SXTn4vi0

After new video driver code.
https://youtu.be/0h4VeYEgU3M

Edit: this does also effect LCD as well.

https://www.youtube.com/watch?v=sXfUYBvLhVk

**Comment by orbea** [comment]
Can you explain the exact problem better? I don't have such a setup making it hard for me to understand fully.

What commit did this start in?

**Comment by alphanu1** [comment]
@orbea When you load latest commit  of RetroArch there is now a delay in it opening on LCD its about 5 times slower. On a CRT weather that be 240p TV or 31khz PC monitor you see the x11 server refresh about 4 times.  Flashing between desktop and RetroArch flickering.

I had not pulled from RetroArch for about 3 weeks so there has been around 200+ commits. I have not been able to narrowed this down to a specific commit yet.

Here is the video showing the introduced delay.

https://www.youtube.com/watch?v=sXfUYBvLhVk

**Comment by orbea** [comment]
You should be able to use `git bisect` to narrow it down further without too much effort. Do you need help learning how to do this?

I also can't seem to reproduce the delay in your new video, it seems the same here with the current master and `v1.7.6`.

**Comment by orbea** [comment]
> Do you need help learning how to do this?

In case you do, this should help.

1. Find a bad commit or commit reference, in this case I think `master` would work?
2. Find a good commit or commit reference, `v1.7.6` is a good commit?
3. Now start the bisect: `git bisect start`
4. Enter a bad commit: `git bisect bad master`
5. Enter a good commit: `git bisect good v1.7.6`
6. It will automatically checkout a commit for you to try now, build it and test it however you normally would.
- If the commit is good: `git bisect good`
- If the commit is bad: `git bisect bad`
- If you encounter an unrelated issue that makes the bisect not possible such a build failure: `git bisect skip`
7. Repeat step 6 until it tells you the first bad commit and share that here.

**Comment by alphanu1** [comment]
OK. So, the first commit that is bad using `git bisect` is commit #8335. This is where the you have the huge delay in opening RA and 240p issues.

There are 7 revisions between the last good commit and this one.

**Comment by orbea** [comment]
@natinusala Any ideas?

**Comment by alphanu1** [comment]
There are 4 commits here. Just going through them now. Commit #8334 does not have the issue.

**Comment by alphanu1** [comment]
@orbea @natinusala 

I have checked these 4 commits and all is OK.
42325ea368227320b2a102eaf3406e98a8d4d42c
6e20d216bde02760de497d0e6807ba252ab29db3
4d86030cc2b3a6473801bd7f6f3588f048c2aa9d
a98cb75549e922420aad872585837776392f9be7

I might need to run `git bisect` again. nit sure what this one had the issue.

However, it is in commit 8984702ba88c6377269d3a9af97d37afa409e188 #8349

**Comment by natinusala** [comment]
So it's not mine ?

I swear I'll be able to make a wig with all the grey hair I grow with those widgets

**Comment by orbea** [comment]
@alphanu1 Could you share the output from `git bisect`, I might be able to figure out where you may be going wrong if I can see more.

You could bisect in one terminal and build in another for convenience or even use two different cloned repos. I would also suggest to run `git clean -xdf` before each build to make sure its a clean state.

**Comment by alphanu1** [comment]
I did not continue through until it gave me the result of the first bad commit.

It is commit 97b7512 more precisely it is the new function that has been added to dispserv_x11.c
`static enum rotation x11_display_server_get_screen_orientation(void)`

I have tested this by remove the code in this function and returning 0 in this commit and the mater branch. The issue is gone. I know this is not the fix but does prove where the issue is.

I can code CRTSwitchRes around this. However, it is still a slow down for users of LCD and an unnecessary video glitch for 240p and CRT PC monitor users.

@bparker06 This is your commit any thoughts on what you want to do here?

**Comment by orbea** [comment]
Thanks for clearing up what commit it was!

**Comment by bparker06** [comment]
If your screen orientation is set to the default "Normal" setting... then the only time that function should even be getting called is at the start of RA or on content load where the video display server interface is inited... and it definitely should not be flashing your screen (it doesn't for me). Only calls to *set*_screen_orientation should do that.

Can you verify which call to get_screen_orientation (from where) is causing your slowdown? And make sure it's not actually calling *set*_screen_orientation anywhere for some reason? The code itself should be *very fast* so I don't understand why you have this problem... but I don't have a CRT setup to be able to verify.

**Comment by alphanu1** [comment]
@bparker06 I though the same, its is strange that it is in get_screen_orientation and not set_screen_orientation. However, this is the code I isolated when I ran the tests. I have not updated to the very latest comment yet, so I will do that soon.

I have worked around it for CRTSwitchRes. So, its not really an issue there anymore. However, it did slow down the load time of RA on my PC with an LCD screen. If this is not an issue for anyone else then I guess we should leave it.

Just to add I am running Lubuntu with gl video driver and an ATI graphics card. Just incase that changes anything.
